lib/config/wrapper.go

```diff
@@ -203,9 +234,17 @@ func (w *wrapper) replaceLocked(to Configuration) (Waiter, error) {
 		}
 	}
 
-	w.cfg = to
+	waiter := w.notifyListeners(from.Copy(), to.Copy())
+	go func() {
```

Comment:
If we return due to an error in any of the conditions above we'll deadlock forever because we increased replaceCounter but currentCounter can never keep up?

(I wonder if this counter-and-cond thing is the best way to represent a config  change queue... I'm thinking a service loop reading from a `chan func(*Config) *Config` that is processed serially, and the config change methods just push their operation to that channel? Make the channel 1000-buffered or something, and if sending to the channel fails it means the config change operation returned an error...)